### PR TITLE
[CSL-1666] Fix demo, get rid of dev-mode

### PR DESCRIPTION
**Pull request by georgeee**
* Fix demo script
* Fix block-gen to accept configuration
* Add `use-genesis-secrets` to `block-gen`

### CODE REVIEW

**Comment by domenkozar** [comment]
You also need to remove dev-mode occurrences in `default.nix`

**Comment by georgeee** [comment]
@domenkozar was done already

**Comment by domenkozar** [comment]
Ah - nevermind.

**Comment by gromakovsky** [comment]
@volhovM it's ready for re-review (I rebased on top of master, also added `--use-genesis-secrets` flag).

File: tools/src/block-gen/Options.hs [hunk 1]
```diff
@@ -19,22 +19,26 @@ import           Options.Applicative          (Parser, auto, execParser, footerD
...
     -- ^ Whether to append to existing db.
-    , bgoSeed        :: !(Maybe Int)
+    , bgoSeed                 :: !(Maybe Int)
     -- ^ Generating seed
-    , bgoTxGenParams :: !TxGenParams
+    , bgoTxGenParams          :: !TxGenParams
     -- ^ Transaction generator parameters
+    , bgoConfigurationOptions :: ConfigurationOptions
+    -- ^ Configuration to run with
+    , bgoUseGenesisSecrets    :: !Bool
```
Comment (volhovM) [inline tools/src/block-gen/Options.hs]: Not documented, but anyway.

**Pull request merged**